packages/x-date-pickers/src/MultiSectionDigitalClock/MultiSectionDigitalClock.tsx

```diff
@@ -396,7 +396,7 @@ export const MultiSectionDigitalClock = React.forwardRef(function MultiSectionDi
       return views;
     }
     const digitViews = views.filter((v) => v !== 'meridiem');
-    const result = digitViews.toReversed();
+    const result: TimeViewWithMeridiem[] = digitViews.toReversed();
```

Comment:
The L401 became a problem after: https://devblogs.microsoft.com/typescript/announcing-typescript-5-5/#inferred-type-predicates
I could have added `as TimeViewWithMeridiem` on the `push` call, but IMHO this is a bit cleaner.
WDYT?

Reply:
Fine for me :ok_hand: